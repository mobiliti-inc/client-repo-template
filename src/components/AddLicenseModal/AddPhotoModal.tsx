import React from 'react';
import Camera from '../Camera/Camera';
import DialogModal from '../Modal/DialogModal/DialogModal';

type AddPhotoModalProps = {
	visible: boolean,
	title: string,
	closeModal: (...args: any[]) => any,
	onBack: (...args: any[]) => any,
	onUsePhoto: (...args: any[]) => any,
	closeCamera: boolean,
	onCameraAccessFail: (...args: any[]) => any
};

type AddPhotoModalState = {
	takePhoto: boolean,
	photoTaken: boolean,
	image: null,
	cameraAccessGranted: boolean
};

class AddPhotoModal extends React.PureComponent<
	AddPhotoModalProps,
	AddPhotoModalState
	> {
	state = {
		takePhoto: false,
		photoTaken: false,
		image: null,
		cameraAccessGranted: false
	};

	onBack = () => {
		if (this.state.photoTaken) {
			this.setState({
				takePhoto: true,
				photoTaken: false
			});
		} else {
			this.props.onBack();
		}
	}

	onCloseModal = () => {
		this.setState({
			takePhoto: false,
			image: null,
			photoTaken: false
		});
		this.props.closeModal();
	}

	onCameraAccessSuccess = () => {
		this.setState({
			cameraAccessGranted: true
		});
	}

	onCameraAccessFail = () => {
		if (this.state.cameraAccessGranted) {
			this.setState({
				cameraAccessGranted: false
			});
		}
		this.props.onCameraAccessFail();
	}

	takePhoto = () => {
		if (this.state.photoTaken) {
			this.setState({
				image: null,
				photoTaken: false
			});
			this.props.onUsePhoto(this.state.image);
		} else {
			// reset in case taking of photo is cancelled
			const { takePhoto } = this.state;
			if (takePhoto) {
				this.setState({
					takePhoto: false
				});
			} else {
				this.setState({
					takePhoto: true
				});
			}
		}
	}

	captureImage = image => {
		this.setState({
			photoTaken: true,
			takePhoto: false,
			image
		});
	}

	render() {
		return (
			<DialogModal
				isBlueButtoDisabled={!this.state.cameraAccessGranted}
				modalIsVisible={this.props.visible}
				modalTitle={this.props.title}
				blueButtonText={this.state.photoTaken ? 'Use Photo' : 'Take Photo'}
				whiteButtonText={this.state.photoTaken ? 'Retake' : 'Back'}
				blueButtonClick={this.takePhoto}
				whiteButtonClick={this.onBack}
				closeModal={this.onCloseModal}
				className="modal-body-class"
			>
				<Camera
					onTakingPhoto={this.captureImage}
					takePhoto={this.state.takePhoto}
					height={300}
					width={500}
					closeCamera={this.props.closeCamera}
					onCameraAccessSuccess={this.onCameraAccessSuccess}
					onCameraAccessFail={this.onCameraAccessFail}
				/>
			</DialogModal>
		);
	}
}
export default CSSModules(AddPhotoModal);
