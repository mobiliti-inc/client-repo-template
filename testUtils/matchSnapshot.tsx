const { shallowToJson } = require('enzyme-to-json');

export const matchSnapshot = (subject: any) => {
	expect(shallowToJson(subject)).toMatchSnapshot();
};
