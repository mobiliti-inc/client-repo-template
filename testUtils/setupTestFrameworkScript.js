const React = require("react");
// const CSSModules = require("react-css-modules");
const Enzyme = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

const { shallow, render, mount } = Enzyme;

Enzyme.configure({ adapter: new Adapter() });
jest.setTimeout(30000);

global.React = React;
// global.CSSModules = CSSModules;
global.shallow = shallow;
global.render = render;
global.mount = mount;
