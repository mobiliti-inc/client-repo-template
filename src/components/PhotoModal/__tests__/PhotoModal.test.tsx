import React from 'react';

import { matchSnapshot } from '../../../../testUtils/matchSnapshot';
import { shallow } from 'enzyme';

import AddPhotoModal from '../PhotoModal';

describe('AddPhotoModal', () => {
	it('renders', () => {
		const subject = shallow(
			<AddPhotoModal
				visible
				title="Front of license"
				closeModal={() => { }}
				onBack={() => { }}
				onUsePhoto={() => { }}
				closeCamera={false}
				onCameraAccessFail={() => { }}
			/>);
		matchSnapshot(subject);
	});
});
