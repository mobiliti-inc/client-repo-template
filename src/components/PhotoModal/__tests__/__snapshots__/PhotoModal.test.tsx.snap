// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPhotoModal renders 1`] = `
<DialogModal
  blueButtonClick={[Function]}
  blueButtonText="Take Photo"
  className="modal-body-class"
  closeModal={[Function]}
  isBlueButtonDisabled={true}
  modalIsVisible={true}
  modalTitle="Front of license"
  whiteButtonClick={[Function]}
  whiteButtonText="Back"
>
  <Camera
    closeCamera={false}
    height={300}
    onCameraAccessFail={[Function]}
    onCameraAccessSuccess={[Function]}
    onTakingPhoto={[Function]}
    takePhoto={false}
    width={500}
  />
</DialogModal>
`;
